### PR TITLE
Add support for GCS storage for pgBackRest

### PR DESCRIPTION
**Pull request by jkatz**
pgBackRest 2.33 introduced support for GCS storage. This adds
supported for GCS storage.

Issue: [ch11177]
CrunchyData/postgres-operator#2368

### CODE REVIEW

File: bin/postgres-ha/pgbackrest/pgbackrest-archive-push-local-s3.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
+
```
Comment (tjmoore4) [inline bin/postgres-ha/pgbackrest/pgbackrest-archive-push-local-s3.sh]: :pick: looks like this space could just be removed?
```suggestion

```

File: conf/pgbackrest-repo/aws-s3-credentials.yaml [hunk 2]
```diff
@@ -1,3 +1,4 @@
 ---
 aws-s3-key:
 aws-s3-key-secret:
+gcs-key:
```
Comment (tjmoore4) [inline conf/pgbackrest-repo/aws-s3-credentials.yaml:4]: :thinking: Similar to the comment on the Operator side, would we want to either rename this file, or possibly create a separate one?
If renaming, perhaps something like 'cloud-storage-credentials.yaml' would be appropriate.

File: build/postgres/Dockerfile [hunk 3]
```diff
@@ -99,6 +99,7 @@ RUN chown -R postgres:postgres /opt/crunchy /var/lib/pgsql \
 # open up the postgres port
 EXPOSE 5432
 
+ADD tmp/pgbackrest /usr/bin/pgbackrest
```
Comment (andrewlecuyer) [inline build/postgres/Dockerfile]: Was this intended? (thinking maybe something from your local pgBackRest build).

**Review by andrewlecuyer — CHANGES_REQUESTED** [review]
Looks great.  Noticed missing executable permissions on one of the new scripts that I was hoping could be adjusted, but otherwise just had a couple of questions around the use of the `pgbackrest.conf` file with `gcs` (not blockers).

File: bin/pgbackrest-repo/pgbackrest-repo.sh [hunk 4]
```diff
@@ -78,7 +78,17 @@ then
 	then
 		printf "repo1-s3-uri-style=%s\n" "${PGBACKREST_REPO1_S3_URI_STYLE}" >> /etc/pgbackrest/pgbackrest.conf
 	fi
-	
+
+  # and GCS variables
+  if [[ "${PGBACKREST_REPO1_GCS_BUCKET}" != "" ]]
+  then
+    printf "repo1-gcs-bucket=%s\n" "${PGBACKREST_REPO1_GCS_BUCKET}" >> /etc/pgbackrest/pgbackrest.conf
+  fi
```
[the remark this answers is not in the document]
Comment (jkatz) [reply bin/pgbackrest-repo/pgbackrest-repo.sh:86]: @andrewlecuyer I was just adding what you add to ensure compatibility. If we don't need it, we don't need it.

File: bin/pgbackrest-repo/pgbackrest-repo.sh [hunk 5]
```diff
@@ -78,7 +78,17 @@ then
...
+
+  if [[ "${PGBACKREST_REPO1_GCS_KEY}" != "" ]]
+  then
+    printf "repo1-gcs-key=%s\n" "${PGBACKREST_REPO1_GCS_KEY}" >> /etc/pgbackrest/pgbackrest.conf
+  fi
```
[the remark this answers is not in the document]
Comment (jkatz) [reply bin/pgbackrest-repo/pgbackrest-repo.sh:91]: See above comment :smile:

**Review by andrewlecuyer — APPROVED** [review]

**Pull request merged**